meeko/molsetup.py

```diff
@@ -4,623 +4,451 @@
-        self.rmsd_symmetry_indices = ()
+        # Tracking atoms and bonds
+        self.atoms: list[Atom] = []
+        self.bond_info = {}
         self.rings = {}
-        self.rings_aromatic = []
-        self.atom_to_ring_id = defaultdict(list)
-        self.ring_corners = {}  # used to store corner flexibility
-        self.name = None
         self.rotamers = []
-
-    def copy(self):
-        newsetup = MoleculeSetup()
-        newsetup.__dict__ = deepcopy(self.__dict__)
-        return newsetup
+        pass
 
     @classmethod
-    def from_mol(cls, mol, keep_chorded_rings=False, keep_equivalent_rings=False,
-                 assign_charges=True):
-        molsetup = cls()
-        molsetup.mol = mol
-        molsetup.atom_true_count = molsetup.get_num_mol_atoms()
-        molsetup.name = molsetup.get_mol_name()
-        molsetup.init_atom(assign_charges)
-        molsetup.init_bond()
-        molsetup.perceive_rings(keep_chorded_rings, keep_equivalent_rings)
-        return molsetup
+    def from_prmtop_inpcrd(cls, prmtop, crd_filename: str):
+        # TODO: pull functionality over and clean up
+        pass
 
-    @classmethod
-    def from_prmtop_inpcrd(cls, prmtop, crd_filename):
-        molsetup = cls()
-        x, y, z = prmtop._coords_from_inpcrd(crd_filename)
-        all_atom_indices = prmtop.vmdsel("")
-        if len(all_atom_indices) != len(x):
-            raise RuntimeError("number of prmtop atoms differs from x coordinates")
-        prmtop.upper_atom_types = prmtop._gen_gaff_uppercase()
-        charges = [chrg/18.2223 for chrg in prmtop.read_flag("CHARGE")]
-        atomic_nrs = prmtop.read_flag("ATOMIC_NUMBER")
-        pdbqt_string = ""
-        molsetup.atom_params["rmin_half"] = []
-        molsetup.atom_params["epsilon"] = []
-        vdw_by_atype = {}
-        for vdw in prmtop._retrieve_vdw():
-            if vdw.atom in vdw_by_atype:
-                raise RuntimeError("repeated atom type from prmtop._retrieve_vdw()")
-            vdw_by_atype[vdw.atom] = {"rmin_half": vdw.r, "epsilon": vdw.e}
-        for i in prmtop.atom_sel_idx:
-            pdbinfo = rdkitutils.PDBAtomInfo(
-                name=prmtop.pdb_atom_name[i],
-                resName = prmtop.pdb_resname[i],
-                resNum = int(prmtop.pdb_resid[i]),
-                chain = " ",
-            )
-            atype = prmtop.upper_atom_types[i]
-            molsetup.add_atom(
-                i,
-                coord = (x[i], y[i], z[i]),
-                charge = charges[i],
-                atom_type = atype,
-                element = atomic_nrs[i], 
-                pdbinfo = pdbinfo,
-                chiral = False,
-                ignore = False,
-                add_atom_params = {
-                    "rmin_half": vdw_by_atype[atype]["rmin_half"],
-                    "epsilon": vdw_by_atype[atype]["epsilon"],
-                }
+    # region Manually Building A MoleculeSetup
+    def add_atom(
```

Comment:
I've implemented the Atom structure mostly so we are not doing a lot more overhead managing indices between attributes/making sure attributes don't get out of sync. With this, we can also define null values for attributes that aren't currently set, or we can leave it as-is. The extent we want the atom structure to be buried varies, I can completely bury it/allow external access to do things like loop over all the elements in index order, if that seems fair.